mslib/mscolab/file_manager.py

```diff
@@ -250,6 +254,42 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def save_user_profile_image(self, user_id, image_file):
+        """
+        Save the user's profile image path to the database.
+        """
+        upload_folder = mscolab_settings.UPLOAD_FOLDER
+        file_token = secrets.token_urlsafe(20)
+
+        with fs.open_fs('/') as home_fs:
+            file_dir = fs.path.join(upload_folder, 'profile')
+            # Checking if directory exists. Create the directory if it does not exist
+            if '\\' not in file_dir:
+                if not home_fs.exists(file_dir):
+                    home_fs.makedirs(file_dir)
+            else:
+                file_dir = file_dir.replace('\\', '/')
+                if not os.path.exists(file_dir):
+                    os.makedirs(file_dir)
+
+            # Creating unique and secure filename
+            file_name, file_ext = image_file.filename.rsplit('.', 1)
```

Comment:
`file_ext` should better be derived from the mime type and the original filename not used at all, to avoid accidentally introducing vulnerabilities like path traversals (ideally I wouldn't want a filename to reach the flask endpoint at all, but I think it is part of flasks design that it is there).